### ISSUE TITLE
Offlineimap sets wrong internaldate

### ISSUE THREAD

**Issue by jgonsior** [OfflineIMAP/offlineimap#522]
Hello,
I'm using Offlineimap, my mail provider is [FastMail](https://www.fastmail.com/). Whenever I'm moving a file around locally (from one Maildir directory into another one) the mail is first being deleted on the server, and then uploaded again, so far so good. But when the mail is uploaded again the internaldate of the message get's changed to the current timestamp:

`2018-02-06 13:28:26 DEBUG: [imap]: savemessage: date: "06-Feb-2018 13:27:33 +0100", […]`

So in the end the mail shows on FastMail with the wrong date. Correct woud be the date which is set in the Date header of the mail. How can I configure Offlineimap to respect the Date header here?

**Comment by nicolas33** [comment]
offlineimap cannot currently set this IMAP optional parameter on APPEND.

**Comment by nicolas33** [comment]
Sorry, I did not remember this feature is supported while not exposed to the user. The internal date is set like this (fallbacks):

1. The timestamp of the local file.
2. The Date header of the email.
3. Let the IMAP server set the internaldate.

For the IMAP side, there isn't a configuration option to tune this but it's easy to introduce one.

**Comment by chris001** [comment]
Yes, this would be great to have.  Note, when the "remote IMAP MOVE method" is added for greater efficiency and true sync accuracy, this `internaldate` will also remain original and accurate - either timestamp of the local file and/or `Date` header of the email message.
